### PR TITLE
Entity Update Flags

### PR DESCRIPTION
**Pull request by broxen**
## Summary
I took a stab at allowing Entity to Update the client only as needed. I did this by creating a integer bit set that can be toggled with flags for the items in EntityUpdateCodec's `serializeto()`.

This PR is built upon #924 SlashCommand Refactoring PR, which should merge first.

### Issues closed
None. But this is a step towards #690 

### Additions/modifications proposed in this pull request:
- EntityUpdateFlags for tracking when updates are necessary to send to the client. This differs from  ClientEntityStateBelief which is the client's understanding of our entity state.
- EntityUpdateFlag helper methods to mark, unmark, and reset the flags for an entity. 
- Lots of minor code cleanups. 
- Removed `GameCommandsList.h` per my note in Discord.

### CODE REVIEW

**Comment by broxen** [comment]
This fails on OSX because... ???
```cpp
/Users/travis/build/Segs/Segs/Projects/CoX/Common/GameData/EntityHelpers.cpp:457:51: error: 'Entity::UpdateFlag::Full' is not a member of class 'Entity'
    e.m_entity_update_flags.setFlag(e.UpdateFlag::Full);
```
It seems it's not recognizing the enum, and that makes no sense to me =P

**Comment by broxen** [comment]
Seems as though the issue is _very technical_ 
https://stackoverflow.com/questions/50027942/why-doesnt-clang-allow-accessing-a-nested-enum-class-through-an-instance

For now I've made it a classic enum and capitalized all the members to avoid name collisions.

File: Projects/CoX/Common/GameData/Movement.cpp [hunk 1]
```diff
@@ -934,27 +936,27 @@ void forcePosition(Entity &e, glm::vec3 pos)
...
-    e.m_update_anims = true;
-    e.m_rare_update = true;
-
     TriggeredMove tmove;
     tmove.m_move_idx = move_idx;
     tmove.m_ticks_to_delay = delay;
     tmove.m_trigger_fx_idx = fx_idx;
 
     e.m_triggered_moves.push_back(tmove);
+    e.m_entity_update_flags.setFlag(e.UpdateFlag::ANIMATIONS);
```
Comment (nemerle) [inline Projects/CoX/Common/GameData/Movement.cpp:959]: To be honest, this looks a bit strange..
usually when referring to a nested enum field you use the full path:
```cpp
ClassName::Enum::Field
```
My advice: 
extract the enum outside of the class, and name it `EntityUpdateFlag`

Comment (broxen) [reply Projects/CoX/Common/GameData/Movement.cpp:959]: Sounds good.

**Pull request merged**